server/src/main/java/org/opensearch/grpc/GrpcTransportSettings.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.grpc;
+
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Setting.Property;
+import org.opensearch.common.transport.PortsRange;
+import org.opensearch.core.common.unit.ByteSizeUnit;
+import org.opensearch.core.common.unit.ByteSizeValue;
+
+import java.util.List;
+import java.util.function.Function;
+
+import static java.util.Collections.emptyList;
+import static org.opensearch.common.settings.Setting.intSetting;
+import static org.opensearch.common.settings.Setting.listSetting;
+
+/**
+ * Transport settings for gRPC connections
+ *
+ * @opensearch.internal
+ */
+public final class GrpcTransportSettings {
```

Comment:
i think you are missing add the grpc settings to clustersettings, otherwise it wont' work with grpc.* in opensearch.yaml.

Reply:
More to that, the GPRC is optional plugin the core should have no knowledge of, should be moved to plugin itself